File: 2026-08/src/day_22/prms_120812/Solution.java
package day_22.prms_120812;

class Solution {
    public int solution(int[] array) {
        int[] cnt = new int[1000];
        for (int x : array) {
            cnt[x]++;
        }

        int ans = -1;
        int max = 0;
        boolean uniq = false;

        for (int i = 0; i < 1000; i++) {
            if (cnt[i] > max) {
                ans = i;
                max = cnt[i];
                uniq = true;
            } else if (cnt[i] == max) {
                uniq = false;
            }
        }

        return uniq ? ans : -1;
    }
}

File: 2026-08/src/day_22/prms_120812/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

int cnt[1000];

int solution(vector<int> array) {
    for (int x : array) cnt[x]++;

    int ans = -1;
    int mx = 0;
    bool uniq = false;

    for (int i = 0; i < 1000; i++) {
        if (cnt[i] > mx) {
            ans = i;
            mx = cnt[i];
            uniq = true;
        } else if (cnt[i] == mx) {
            uniq = false;
        }
    }

    return uniq ? ans : -1;
}
